src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentExcerptService.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// excerpt some part of <see cref="Document"/>
+    /// </summary>
+    internal interface IDocumentExcerptService : IDocumentService
+    {
+        /// <summary>
+        /// return <see cref="ExcerptResult"/> of given <see cref="Document"/> and <see cref="TextSpan"/>
+        /// 
+        /// the result might not be an exact copy of the given source or contains more then given span
```

Comment:
my wording might be bad. input is generated C# files, and output is original content where the generated files are generated.

wanted to say that result might not from the generated C# files. this abstraction doesn't have assumption on the implemetation detail/relationship between the original content (such as razor, xaml, resx and etc) and generated files (C# g.i.cs or secondary buffer, or .designer.cs files).

it assumes that is a block box and the provider knows the mapping since the document is from the provider's DynamicFileInfo.